### ISSUE TITLE
[a11y Design Review] MHV Cartography - Homepage Team: Making Health Contacts Editable in Profile - Figma

### ISSUE THREAD

**Issue by rileyorr** [department-of-veterans-affairs/va.gov-team#92319]
> [!NOTE]
> **Current Phase:** The team is currently in the ideation phase designing how Veterans might edit and remove **Profile** health contacts and is looking for accessibility feedback as they explore different ideas. 

## Issue Description
The MHV homepage team (Cartography) has existing functionality on VA.gov's Profile page that displays read-only content containing the Veteran's primary and secondary (if applicable) emergency contacts as well as primary and secondary (if applicable) next of kin contacts. 

Research has found that Veterans expect to be able to edit and remove emergency contacts and next of kin contacts from within their Profile rather than needing to contact a phone number to do so. 

### Supporting Artifacts 
- [Figma file](https://www.figma.com/design/uYMdNtQnrzq7OiDl0rPF9k/Profile---Personal-health-care-contacts?node-id=0-444&t=b0yUiAMPZ5Q76Ppe-1) containing design explorations 
- [Mural](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1725485954256/6e1203260ff93e83c7960595f3ce7d3d194094d0?sender=u24af215c7d9bf806b8545488) containing flows and early IA feedback 

```[tasklist]
### a11y Tasks
- [x] Collaborate with Jonathan both asynchronously and synchronously as needed to provide feedback and recommendations as designs are being fleshed out
- [x] Document recommendations and feedback in this ticket so it can be referenced in the future as needed
- [x] **Coverage:** Jamie to assist while Riley is OOO week of 9/9/2024 with any questions Jonathan may have. Ref. comments in this ticket and slack channel [#mhv-on-vagov-cartography-team](https://dsva.slack.com/archives/C0581MN69TJ)
```
```[tasklist]
### Acceptance Criteria
- [x] Feedback on final design(s) are provided
```
```[tasklist]
## Ticket Updates (CAIA Internal)
- [x] _Connect to an `Epic` or `Intake` (what body of work is this a part of?)_ tracked with `caia-mhv`
- [x] _Label with `Originator/Team` (product team or stakeholder requesting support)_
- [x] _Label date in the `Open Date` field_
- [x] _Label with `Estimate` (level of effort expected for this ticket)_
- [x] _Add `Assignee(s)` name(s) to ticket_ (include Naomi)
- [x] _Add `Assignee(s) name(s) to each task` they will complete via handle tag (if known)_
- [x] _Select a `Priority Level`_
- [x] _Update date in `Last Checked` field_
- [x] _Label with `Actual` (level of effort it took to complete this ticket)_
- [x] _Update date in_ `Closed Date`
```

**Comment by rileyorr** [comment]
Here's some notes from my meeting with Jonathan yesterday about this work: 
- Jonathan walked me through what the current read-only experience is on VA.gov and showed me other areas of Profile where Veterans can edit and/or remove contact information 
- He walked through multiple design options in Figma ranging from 1 emergency contact and/or 1 next of kin contact to 2 of each and how they might edit/remove these contacts
 
- The main funkiness comes in when the Veteran wants to remove their primary contact because currently Veterans aren't required to have a primary contact listed: 
   - Should they indicate whether or not they want their secondary contact to get bumped up to primary? 
   - Should the secondary contact be automatically bumped up to primary since the secondary is contacted anyway if the primary is unable to be contacted?
   - The same likely applies to next of kin 
- **Recommendation:** Work with product to decide if secondary contacts should be promoted to primary if the primary is removed. Removing the primary contact, leaving an empty state that prompts Veterans to add a primary but also having content under the secondary contact notifying Veterans that the secondary contact will be reached out to if the primary is unreachable introduces cognitive complexity that may be confusing for Veterans who both do and do not have cognitive disabilities. 

- The current 'Remove' functionality pops up a modal that prevents the Veteran from completing a destructive task without confirming that they do want to remove the contact. There were explorations of adding a checkbox to bump the secondary to primary, but since we know folks often scan content they may miss the checkbox entirely. I would not recommend a multi-step modal to split into a 'one question per page' flow. 
   - The profile's notification page does contain a flow that may work in this instance. For example, when the Veteran clicks the edit link to edit their phone number they're taken to a separate page that appears similarly to our VA form UI. They update their information on that page and select a 'Save' button. A similar approach could apply here where the Veteran confirms they want to remove their primary contact and the next page shows the secondary contact information and asks if they want to make that the primary contact. That still doesn't solve the potential cognitive challenges if they select no and the primary contact is left as an empty state.

- In other parts of Profile if an optional data field is in an empty state there's content that tells the Veteran to select the 'Edit' button to add a new contact.
   - It would be worth exploring the background of this pattern because it seemed confusing to both me and Jonathan. We both thought we would expect the button to read something like 'Add a next of kin' or 'Add a contact'. 
   - **Recommendation:** Test the existing functionality with a screen reader to hear what's read when navigating to the 'Edit' button on an empty state contact. It could be that updating the button to something with the word 'Add' might better align with [SC 2.4.4 Link Purpose](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html) since folks' mental models for 'Edit' assumes something is there to change or update, whereas 'Add' implies there is nothing currently there. 

- We talked about potentially removing the H3s that indicate primary and secondary contacts but decided to keep them after talking through how some folks navigate by heading. There would also need to be an indication if there is more than one emergency contact that there's 1 of 2 or 2 of 2 in the list. Removing the headers also doesn't solve the main funkiness of the flow. 

**Next steps:** I am waiting for followup from Jonathan to continue collaboration on the designs.

**Comment by sara-amanda** [comment]
@coforma-jamie adding you to this ticket for the week of 9/9/2024, based on Riley's coverage doc. See task list for additional information and comment in this ticket. We appreciate it!

**Comment by jonathanjnelson** [comment]
@rileyorr Curious if you think this is too much information to collect inline?[ You can see the flow in Figma ](https://www.figma.com/design/uYMdNtQnrzq7OiDl0rPF9k/Profile---Personal-health-care-contacts?node-id=2187-5640&t=8amv49yFqatJGRTQ-1)

![Image](https://github.com/user-attachments/assets/5830245d-b60b-477a-b181-ff962acee452)

**Comment by jonathanjnelson** [comment]
Another option might be this edit as a subtask flow. [Flow in Figma here](https://www.figma.com/design/uYMdNtQnrzq7OiDl0rPF9k/Profile---Personal-health-care-contacts?node-id=0-444&t=8amv49yFqatJGRTQ-1) 

But I am wondering if this is over designing/engineering the experience? Would we need a review page? I would think the fact that the Veteran can edit if there is a mistake but curious about your thoughts!

**Comment by rileyorr** [comment]
@jonathanjnelson these designs look great! I think either solution could work, so I'll provide feedback on both. 

### Option 1: Inline updates 
[Figma flow I'm referencing](https://www.figma.com/design/uYMdNtQnrzq7OiDl0rPF9k/Profile---Personal-health-care-contacts?node-id=2187-5640&t=8amv49yFqatJGRTQ-1)

The case could be made for displaying all inputs for a contact at once because there could be multiple emergency contacts and next of kin contacts. This pattern would allow Veterans to quickly update multiple pieces of information at once or quickly remove a contact without needing to tab too much.

I know a best practice is to break forms into chunks to reduce cognitive load, but I'm not sure that pattern would be beneficial in this case. The profile's personal information section uses a design that allows Veterans to update inline information by section, but this type of solution may not work in this case for a couple reasons: 
- If Veterans expect to update multiple sections of information at once (ex. next of kin phone number and address) this adds extra steps and tabs to get to update multiple sections 
- If the Veteran has multiple emergency contacts and next of kins, the page would become extremely long (especially on mobile) and require many tabs to get to information to update 
- The flow to remove a contact may become complex 

Example of the personal information flow for reference: 

https://github.com/user-attachments/assets/ade81b3f-cac8-44e1-b65f-b607cd323908

If your team decides to move forward with the inline updates design, something to consider is what happens if a Veteran:
- Begins to update a contact and selects the 'Cancel' button 
- Begins to update a contact and navigates to a different contact 

To prevent destructive actions, such as data loss, from occurring we should notify the Veteran that their changes won't be saved and to confirm they want to cancel. Or alert them to save their updated information before they're able to update another contact. 

### Option 2: Subtask flow 

I think the subtask flow idea may work well in this case because the Veteran would be updating all the emergency contact's information at once: name, address, phone number. And this would allow you to break down the form into more digestible chunks to reduce cognitive overload. 

You would not need to have a review page in this case because according to WCAG [SC 3.3.6 Error Prevention (All)](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-all.html) and [SC 3.3.4 Error Prevention (Legal, Financial, Data)](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html) **one of the following** must be true: 
- Submissions are reversible 
- Data entered by the user is checked for input errors and the user is provided an opportunity to correct them
- A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission

Submissions would be reversible because the Veteran could immediately edit the contact information to what was there previously, and the data would be checked for input errors. You could provide a review page prior to submission so the Veteran could see the entire contact's information prior to submission, but I don't believe it's a requirement per the SC provided above. However, this is just a recommendation and you may get other feedback from the governance team. 

### Miscellaneous 
I've provided a recording below of both the existing Contact Information and Notification flows which include inline updates and subtask updates to compare. As you can see, the inline updates may be more complicated when considering the complexity of allowing the Veteran to update all contacts in one view if they attempt to update other contacts.

https://github.com/user-attachments/assets/9f443b27-7b00-43e3-9913-759fa0118cd8

Some things to consider: 
- How often are Veterans updating a single piece of information for a contact vs multiple pieces of information? 
- Is there any usability research related to the Notification subtask flow and the Contact Information inline flow that may uncover any usability issues with one pattern vs the other? 

I know this is a lot of feedback so lmk if you have any questions or you want to hop on a call to chat through anything! Great work on this, I'm excited to continue collaborating and seeing where this goes!

**Comment by rileyorr** [comment]
Hi @jonathanjnelson! I noticed that the ticket related to updating the designs (#93138) is now closed since the designs have been updated and the work will be picked back up once the team is ready for implementation. 

Do you need additional accessibility support for this work at this time? If not, I can go ahead and start the process to close this ticket so the design review work is captured on CAIA's end. We can open another ticket in the future to capture additional accessibility support when your team is ready to implement the designs.

**Comment by rileyorr** [comment]
Confirmed with Jonathan that this ticket is good to close! 

cc @sara-amanda @NaomiPMC
